emcc.py

```diff
@@ -2809,8 +2823,9 @@ def get_full_import_name(name):
     # JS, you may need to manipulate the refcount manually not to leak memory.
     # What you need to do is different depending on the kind of EH you use
     # (https://github.com/emscripten-core/emscripten/issues/17115).
-    settings.DEFAULT_LIBRARY_FUNCS_TO_INCLUDE += ['$getExceptionMessage', '$incrementExceptionRefcount', '$decrementExceptionRefcount']
-    settings.EXPORTED_FUNCTIONS += ['getExceptionMessage', '___get_exception_message', '_free']
+    if not settings.DISABLE_EXCEPTION_CATCHING or settings.WASM_EXCEPTIONS:
```

Comment:
Issuing an error sounds good to me. This condition was to avoid these linker errors:
```console
$ echo -e 'int main(){return 0;}' | ./emcc -xc -sEXPORT_EXCEPTION_HANDLING_HELPERS -o test.js -
error: undefined symbol: $decrementExceptionRefcount (referenced by top-level compiled C/C++ code)
warning: To disable errors for undefined symbols use `-sERROR_ON_UNDEFINED_SYMBOLS=0`
warning: decrementExceptionRefcount may need to be added to EXPORTED_FUNCTIONS if it arrives from a system library
error: undefined symbol: $getExceptionMessage (referenced by top-level compiled C/C++ code)
warning: getExceptionMessage may need to be added to EXPORTED_FUNCTIONS if it arrives from a system library
error: undefined symbol: $incrementExceptionRefcount (referenced by top-level compiled C/C++ code)
warning: incrementExceptionRefcount may need to be added to EXPORTED_FUNCTIONS if it arrives from a system library
Error: Aborting compilation due to previous errors
```

I could make such a change in this PR, but it would probably be neater to make a separate PR for this.

Reply:
I tried that in #18696.

Reply:
I updated #18696 to tidy up the relationship between `ASSERTIONS`, `EXCEPTION_STACK_TRACES`, and `EXPORT_EXCEPTION_HANDLING_HELPERS`. That includes "`EXCEPTION_STACK_TRACES` implies `EXPORT_EXCEPTION_HANDLING_HELPERS` part" in this PR, so if that lands first, this PR might need to fix a few conflicts.